### PR TITLE
Refactor the docs

### PR DESCRIPTION
**Pull request by arcondello**
This was a basic refactor - I have not changed much of the content. A more thorough walkthrough is still needed.

### CODE REVIEW

File: docs/README.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+/home/arcondello/projects/ocean/dwave-samplers/README.rst
```
Comment (JoelPasvolsky) [inline docs/README.rst:1]: I'll add this comment outside the PR

Comment (arcondello) [reply docs/README.rst:1]: right, will fix.

File: README.rst [hunk 2]
```diff
@@ -1,2 +1,176 @@
...
+
+Simulated Annealing
+~~~~~~~~~~~~~~~~~~~
+
+`Simulated annealing <https://en.wikipedia.org/wiki/Simulated_annealing>`_ can be used for approximate Boltzmann sampling or heuristic optimization. This implementation approaches the equilibrium distribution by performing updates at a sequence of increasing beta values, beta_schedule, terminating at the target beta. Each spin is updated once in a fixed order per point in the beta_schedule according to a Metropolis- Hastings update. When beta is large the target distribution concentrates, at equilibrium, over ground states of the model. Samples are guaranteed to match the equilibrium for long 'smooth' beta schedules.
+
+>>> from dwave.samplers import SimulatedAnnealingSampler
+>>> sampler = SimulatedAnnealingSampler()
+
+Get a random binary quadratic model (BQM)
```
Comment (JoelPasvolsky) [inline README.rst:41]:
```suggestion
Create a random binary quadratic model (BQM).
```

File: README.rst [hunk 3]
```diff
@@ -1,2 +1,176 @@
...
+
+>>> import dimod
+>>> bqm = dimod.generators.gnp_random_bqm(100, .5, 'BINARY')
+
+Sample using simulated annealing, both with the default beta range and a user-specified one
```
Comment (JoelPasvolsky) [inline README.rst:46]:
```suggestion
Sample using simulated annealing, both with the default beta range and a user-specified one.
```

File: README.rst [hunk 4]
```diff
@@ -1,2 +1,176 @@
...
+>>> from dwave.samplers import SteepestDescentSolver
+>>> solver = SteepestDescentSolver()
+
+Construct the QUBO:
+
+>>> from dimod import Binaries
+>>> x, y = Binaries(['x', 'y'])
+>>> qubo = x + y - 2.5 * x * y
+
+If the solver starts up hill from the global minimum, it will take the
```
Comment (JoelPasvolsky) [inline README.rst:69]:
```suggestion
If the solver starts uphill from the global minimum, it takes the
```

File: README.rst [hunk 5]
```diff
@@ -1,2 +1,176 @@
...
+steepest path, thereby finding the optimal solution.
+
+>>> sampleset = solver.sample(qubo, initial_states={'x': 0, 'y': 1})
+>>> print(sampleset)
+   x  y energy num_oc. num_st.
+0  1  1   -0.5       1       1
+['BINARY', 1 rows, 1 samples, 2 variables]
+
+If the solver starts in a local minima, it will get stuck
```
Comment (JoelPasvolsky) [inline README.rst:78]:
```suggestion
If the solver starts in a local minimum, it gets stuck.
```

File: README.rst [hunk 6]
```diff
@@ -1,2 +1,176 @@
...
+
+Tree Decomposition
+~~~~~~~~~~~~~~~~~~
+
+`Tree decomposition <https://en.wikipedia.org/wiki/Tree_decomposition>`_-based solvers have a runtime that is exponential in the `treewidth <https://en.wikipedia.org/wiki/Treewidth>`_ of the problem graph. For problems with low treewidth, the solver can find ground states very quickly. However, the performance is very poor for even moderately dense problems.
+
+>>> from dwave.samplers import TreeDecompositionSolver
+>>> solver = TreeDecompositionSolver()
+
+Construct a large, tree-shaped problem
```
Comment (JoelPasvolsky) [inline README.rst:121]:
```suggestion
Construct a large, tree-shaped problem.
```

File: dwave/samplers/sa/sampler.py [hunk 7]
```diff
@@ -25,19 +25,33 @@
...
 class SimulatedAnnealingSampler(dimod.Sampler, dimod.Initialized):
     """Simulated annealing sampler.
 
+    An implementation of a simulated annealing sampler.
+
+    A simulated annealing sampler can be used for approximate Boltzmann sampling or
+    heuristic optimization. This implementation approaches the equilibrium
+    distribution by performing updates at a sequence of increasing beta values,
+    ``beta_schedule``, terminating at the target beta. Each spin is updated once
+    in a fixed order per point in the beta_schedule according to a Metropolis-
```
Comment (JoelPasvolsky) [inline dwave/samplers/sa/sampler.py:40]:
```suggestion
    in a fixed order per point in the ``beta_schedule`` according to a Metropolis-
```

**Review by JoelPasvolsky — APPROVED** [review]

**Pull request merged**